components/calendar/src/calendar.rs

```diff
@@ -19,6 +19,16 @@ use core::fmt;
 pub trait Calendar {
     /// The internal type used to represent dates
     type DateInner: PartialEq + Eq + Clone + fmt::Debug;
+    /// Construct a date from era/month codes and fields
+    fn date_from_codes(
+        &self,
+        era: types::Era,
+        year: i32,
+        month_code: types::MonthCode,
+        day: u8,
+    ) -> Result<Self::DateInner, DateTimeError> {
+        Err(DateTimeError::OutOfRange)
```

Comment:
Nit: Doesn't seem like the best error name

Reply:
Yeah, it's preexisting. I considered cleaning up the errors but it might be better to do that separately. We can change the errors in the future, either way.